assets/source/setup-guide/app/helpers/get-context-by-path.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * External dependencies
+ */
+import { getPath } from '@woocommerce/navigation';
+
+/**
+ * Returns the formatted context based on the path.
+ *
+ * @return {string} The path in snake case format
+ * @throws Exception if getPath is not working, returns an empty string as a result
+ */
+export default function getContextByPath() {
```

Comment:
:question: / :construction: 
I don't really get the reason **why** we need this function?

So far for the `context` property, we were hard coding identifiers, so we can get more granular than `path`, and have multiple triggers on the same page and on the same page.
For the triggers that need path, we pass it as is `/with/stashes`.

IMHO, context constructed as path, introduces quite an inconsistency in our Tracks.

BTW, for the `path` property, I'd rather use routing path than the query location path. But that's a separate topic.

Also, the logic with returning `''` is unclear to me, when it may happen, what does it mean?

Reply:
Fixed in https://github.com/woocommerce/pinterest-for-woocommerce/pull/295/commits/e51fbfd87d6b2a2e016196ee9ae73a2248a399b3

> I don't really get the reason why we need this function?

The idea as the beginning was to use the path as a context builder to preserve consistency. [After discussion ](p1639630485065000-slack-C02B84SRR5M) we decided to manually hardcoded the names and the function was considered a bad idea at the end. 

> Also, the logic with returning '' is unclear to me, when it may happen, what does it mean?

Since `getPath` is an external function and `context` is a string,  I decided that, in case it fails,  we could return an empty string instead of `null`, `undefined` or `false`.

Reply:
Updated here, https://github.com/woocommerce/pinterest-for-woocommerce/pull/295/commits/a2d7a4c93491393ebde99e0fba62c0f840831037

We don't really need to pass any view. Since `pinterest_connection` is not being used as a proper naming. In the parent component that refers again and again to 'settings' in the views.

I refer again to #302 for going further on this

Let e know your thoughts about including the view `pinterest_connection`/`pinterest_settings` or only `pinterest_settings`